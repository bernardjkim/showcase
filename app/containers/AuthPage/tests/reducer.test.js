import { fromJS } from 'immutable';
import AuthPageReducer from '../reducer';
import {
  createToken,
  createTokenError,
  createTokenSuccess,
  createUser,
  createUserError,
  createUserSuccess,
} from '../actions';

describe('AuthPageReducer', () => {
  let state;
  beforeEach(() => {
    state = fromJS({
      loading: false,
      error: false,
    });
  });

  it('returns the initial state', () => {
    expect(AuthPageReducer(undefined, {})).toMatchSnapshot();
  });
  it('handle the create token action', () => {
    const email = 'email';
    const password = 'password';
    expect(
      AuthPageReducer(state, createToken(email, password)),
    ).toMatchSnapshot();
  });

  it('handle the create token success action', () => {
    expect(AuthPageReducer(state, createTokenSuccess())).toMatchSnapshot();
  });

  it('handle the create token error action', () => {
    const error = new Error('Test error');
    expect(AuthPageReducer(state, createTokenError(error))).toMatchSnapshot();
  });

  it('handle the create user action', () => {
    const username = 'username';
    const email = 'email';
    const password = 'password';
    const confirmPassword = 'password';
    expect(
      AuthPageReducer(
        state,
        createUser(username, email, password, confirmPassword),
      ),
    ).toMatchSnapshot();
  });

  it('handle the create user success action', () => {
    expect(AuthPageReducer(state, createUserSuccess())).toMatchSnapshot();
  });

  it('handle the create user error action', () => {
    const error = new Error('Test Error');
    expect(AuthPageReducer(state, createUserError(error))).toMatchSnapshot();
  });
});
