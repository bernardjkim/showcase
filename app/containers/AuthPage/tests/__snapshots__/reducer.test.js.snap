// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthPageReducer handle the create user action 1`] = `
Immutable.Map {
  "loading": true,
  "error": false,
}
`;

exports[`AuthPageReducer handle the create user error action 1`] = `
Immutable.Map {
  "loading": false,
  "error": [Error: Test Error],
}
`;

exports[`AuthPageReducer handle the create user success action 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
}
`;

exports[`AuthPageReducer returns the initial state 1`] = `
Immutable.Map {
  "loading": false,
  "error": false,
}
`;
